Story/webclient to restclient (#738)
valtimo-platform/valtimo-documentation: release-notes/major12/12.3.0/valtimo-backend-libraries.md

    
    Constructors in affected classes now use RestClient.Builder instead of WebClient.Builder.
    
    The default HTTP client is now Apache Client v5.x.
    For more details, see [ApacheRequestFactoryCustomizer](https://github.com/valtimo-platform/valtimo-backend-libraries/blob/b0941b2ca161601094203b38ac639f1a809988a8/contract/src/main/kotlin/com/ritense/valtimo/contract/client/ApacheRequestFactoryCustomizer.kt).
    See also [Spring docs](https://docs.spring.io/spring-boot/reference/io/rest-client.html#io.rest-client.restclient.customization)
    
    A new configuration class has been introduced to modify two main timeout settings.
    For more information, see [ValtimoHttpRestClientConfigurationProperties](https://github.com/valtimo-platform/valtimo-backend-libraries/blob/b0941b2ca161601094203b38ac639f1a809988a8/contract/src/main/kotlin/com/ritense/valtimo/contract/client/ValtimoHttpRestClientConfigurationProperties.kt).
    
    Logging with the new Apache client can be configured as follows:
    ```yaml
      logging:
          level:

>               org.apache.http: DEBUG

Reviewer: ```suggestion
              org.apache.http: DEBUG
              com.ritense.valtimo.web.logging.LoggingRestClientCustomizer: DEBUG
```

The following was deprecated:

* **WebClient is no longer the default HTTP client**
    

>     Transitioned from using the WebClient class to RestClient. For more details, see the [overview](https://docs.spring.io/spring-framework/reference/integration/rest-clients.html).

Reviewer 2: ```suggestion
    Transitioned from using the WebClient class to RestClient. For information on this client, see the [Spring documentation](https://docs.spring.io/spring-framework/reference/integration/rest-clients.html).
```

    
    The primary reason for this switch is to adopt a simpler client that does not rely on reactive programming concepts. Most calls in Valtimo require synchronous execution, making WebClient less suitable.
    

>     Constructors in affected classes now use RestClient.Builder instead of WebClient.Builder.

Reviewer 2: ```suggestion
    Constructors in affected classes now use `RestClient.Builder` instead of `WebClient.Builder`.
```

    
    The default HTTP client is now Apache Client v5.x.
    For more details on how to override/adjust this, see [ApacheRequestFactoryCustomizer](https://github.com/valtimo-platform/valtimo-backend-libraries/blob/b0941b2ca161601094203b38ac639f1a809988a8/contract/src/main/kotlin/com/ritense/valtimo/contract/client/ApacheRequestFactoryCustomizer.kt).

>     See also [Spring docs](https://docs.spring.io/spring-boot/reference/io/rest-client.html#io.rest-client.restclient.customization)

Reviewer 2: ```suggestion
    See also [Spring documentation](https://docs.spring.io/spring-boot/reference/io/rest-client.html#io.rest-client.restclient.customization).
```

>     A new configuration class has been introduced to modify two main timeout settings.

Reviewer 2: Since there's a new way of doing things, we should also provide proper documentation for that. I'm thinking this should go in the reference section, explaining what properties can be configured to set up your rest client. What do you think?

    For more information, see [ValtimoHttpRestClientConfigurationProperties](https://github.com/valtimo-platform/valtimo-backend-libraries/blob/b0941b2ca161601094203b38ac639f1a809988a8/contract/src/main/kotlin/com/ritense/valtimo/contract/client/ValtimoHttpRestClientConfigurationProperties.kt).
    
    Logging with the new Apache client can be configured as follows:
    ```yaml
      logging:
          level:
              org.apache.http: DEBUG
    ```
    Additionally, there is a new class for logging requests and responses: LoggingRestClientCustomizer
*  ```yaml
      logging:
          level:
              com.ritense.valtimo.web.logging.LoggingRestClientCustomizer: DEBUG
   ```

* **Methods inside ClientTools**

  Use of WebClient is deprecated, this was used before:

>   - fun <T> getTypedPage(responseClass: Class<out T>): ParameterizedTypeReference<Page<T>> {

Reviewer 2: ```suggestion
  - `fun <T> getTypedPage(responseClass: Class<out T>): ParameterizedTypeReference<Page<T>> {`
```

>   - fun zgwErrorHandler(): ExchangeFilterFunction {

Reviewer 2: ```suggestion
  - `fun zgwErrorHandler(): ExchangeFilterFunction {`
```
